src/components/MultiselectProblem.tsx

```diff
@@ -1,7 +1,120 @@
-import { BaseProblemProps } from './ProblemSetBlock'
+import { useCallback, useState } from 'react'
+import { mathifyElement } from '../lib/math'
+import { BaseProblemProps, NO_MORE_ATTEMPTS_MESSAGE } from './ProblemSetBlock'
+import { Formik, Field, Form, ErrorMessage } from 'formik'
+import * as Yup from 'yup'
 
-interface MultiSelectProps extends BaseProblemProps {}
+interface MultiselectProps extends BaseProblemProps {
+  solutionOptions: string
+}
+
+interface MultiselectFormValues {
+  response: string[]
+}
+
+export const MultiselectProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback, content, buttonText, solutionOptions,
+  correctResponse, encourageResponse, solution, retryLimit
+}: MultiselectProps): JSX.Element => {
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const solutionArray: string[] = JSON.parse(solution)
+  const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+  const schema = Yup.object({
+    response: Yup.array().min(1, 'Please select an answer')
+  })
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const modifyModel = (values: MultiselectFormValues, e: React.ChangeEvent<HTMLInputElement>): string[] => {
+    const newSet = new Set(values.response)
+    if (e.target.checked) {
+      newSet.add(e.target.value)
+    } else {
+      newSet.delete(e.target.value)
+    }
+    return Array.from(newSet)
+  }
+
+  const generateOptions = (values: MultiselectFormValues, isSubmitting: boolean, setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): JSX.Element[] => {
+    const options: JSX.Element[] = []
+
+    parsedOptionValues.forEach(val => options.push(
+      <>
```

Comment:
instead of empty here, you can do `<div className="form-check">` to get the styling to look right (vs applying the style to the problem below). Also, you'll then want to lift `key={val}` here (it needs to go in the outermost element). You can check / confirm for the errors in the console to be sure these are working properly.

Reply:
@mjoss17 - I think you may have missed the second half of this comment about moving `key={val}` to the new `<div className="form-check">` line you added. Otherwise right now you still get the following React gripe:

![image](https://user-images.githubusercontent.com/839025/168836047-50135a52-558e-4bae-aa9e-ae16e3329feb.png)


If you can fix :point_up: real quick I think everything else looks great!

Reply:
Thank you, Ah!